lstchain/mc/sensitivity.py

```diff
@@ -68,124 +73,279 @@ def process_mc(simtelfile, dl2_file):
     sim_par = read_sim_par(source)
     events = pd.read_hdf(dl2_file)
 
-    e_reco = 10**events.mc_energy * u.GeV
-    # n_reco = e_reco.shape[0]
+    e_reco = 10**events.mc_energy.to_numpy() * u.GeV
     gammaness = events.gammaness
-    theta2 = (events.src_x - events.src_x_rec)**2 + \
-        (events.src_y - events.src_y_rec)**2 * u.deg**2
 
-    return gammaness, theta2, e_reco, sim_par
+    #Get source position in radians
+
+    focal_length = source.telescope_descriptions[1]['camera_settings']['focal_length'] * u.m
+
+
+    # If the particle is a gamma ray, it returns the squared angular distance
+    # from the reconstructed gamma-ray position and the simulated incoming position
+    if mc_type=='gamma':
+        events = events[events.mc_type==0]
+        alt2 = events.mc_alt
+        az2 = np.arctan(np.tan(events.mc_az))
+
+    # If the particle is not a gamma-ray (diffuse protons/electrons), it returns
+    # the squared angular distance of the reconstructed position w.r.t. the
+    # center of the camera
+    else:
+        events = events[events.mc_type!=0]
+        alt2 = events.mc_alt_tel
+        az2 = np.arctan(np.tan(events.mc_az_tel))
+
+    src_pos_reco = reco_source_position_sky(events.x.values * u.m,
+                                            events.y.values * u.m,
+                                            events.disp_dx_rec.values * u.m,
+                                            events.disp_dy_rec.values * u.m,
+                                            focal_length,
+                                            events.mc_alt_tel.values * u.rad,
+                                            events.mc_az_tel.values * u.rad)
+
+    alt1 = src_pos_reco.alt.rad
+    az1 = np.arctan(np.tan(src_pos_reco.az.rad))
+
+    angdist2 = (angular_separation(az1, alt1, az2, alt2).to_numpy() * u.rad)**2
```

Comment:
Yes, it works because actually alt1, az1, alt2, az2 are Series objects (from the pandas dataframe) so the result from angular_separation won't conserve any unit.

Reply:
Ho yes you are right, my bad.